### PR TITLE
XP-248 - Use structlog for logging

### PR DESCRIPTION
**Pull request by tanertopal**
### References
Resolves XP-248

### Summary
Replaces current logging mechanism with structlog

### CODE REVIEW

**Comment by atymoshchuk** [comment]
@tanertopal Please use full github template for every pull request.

File: xain_fl/config.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 def init_config():
     """Creates initial config file if non exists"""
-    logger.info("Initializing config in %s", config_file)
+    logger.info(f"Initializing config in {config_file}")
```
Comment (atymoshchuk) [inline xain_fl/config.py]: Please do not use f-strings in logs, it's a rule in python, according to https://www.python.org/dev/peps/pep-0498/

File: xain_fl/datasets/storage.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def fetch_ndarray(url, fpath):
     """
     r = requests.get(url, stream=True)
 
-    logger.info("Fetching file %s", url)
+    logger.info(f"Fetching file {url}")
```
Comment (atymoshchuk) [inline xain_fl/datasets/storage.py]: benefit of structlog is that it allows to use variables key=value in logs:
`logger.info("Fetching file", url=str(url))`
Please adjust log messages accordingly.

**Comment by tanertopal** [comment]
Closing as the branch name ist not accepted. Will open another PR.

**Pull request closed without merging**